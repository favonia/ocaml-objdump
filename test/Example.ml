let test x = Format.printf "%a@." Objdump.pp x

type t = A | B of int | C

type _ u = ..
type _ u += D : int u
type _ u += E : 'a -> 'a u

type v = ..
type v += U : 'a u -> v

let () = test ()
let () = test A
let () = test (B 100)
let () = test C
let () = test 42
let () = test false
let () = test (-1)
let () = test None
let () = test (Some (Some (Some (Some (Some (Some (Some (Some 10))))))))
let () = test (Failure "str")
let () = test (Invalid_argument "str")
let () = test "a very very very very very very very long string with \x012 and \" and ' and \\"
let () = test (fun () -> 1)
let () = test ((("fst","snd"),2,3.0,(),(fun () -> ())))
let f x = (42, fun () -> x)
let () = test (f 1000)
let () = test 3.14
let () = test (2.18, "a")
let () = test (U D)
let () = test (U (E 10))
let () = test (lazy (1 + 2))
let () = let m = lazy (1 + 2) in let _ = Lazy.force m in test m
let () = let rec m = lazy (test m) in Lazy.force m

let rec f x = g x
and g x = f x

let () = test f
let () = test g
let () = test [|1; 2; 3|]
let () = test [|1.0; 2.0; 3.0|]
let () = test [1; 2; 3]

type _ Effect.t += F : unit Effect.t
let k = Effect.Deep.try_with Effect.perform F
    { effc = fun (type a) (e : a Effect.t) ->
          match e with
          | F -> Some (fun (k : (a, _) Effect.Deep.continuation) -> test k; Effect.Deep.continue k ())
          | _ -> None }
