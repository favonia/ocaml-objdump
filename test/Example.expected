int(0)
int(0)
variant0(int(100))
int(1)
int(42)
int(0)
int(-1)
int(0)
variant0(
  variant0(
    variant0(variant0(variant0(variant0(variant0(variant0(int(10)))))))))
variant0(object{class=string("Failure"),object-id=int(-3)},string("str"))
variant0(
  object{class=string("Invalid_argument"),object-id=int(-4)},
  string("str"))
string(
  "a very very very very very very very long string with \0012 and \" and ' and \\")
closure{code=<0x55920bc99390>,arity=1,env=()}
variant0(
  variant0(string("fst"),string("snd")),
  int(2),
  double(3.000000),
  int(0),
  closure{code=<0x55920bc993a0>,arity=1,env=()})
variant0(int(42),closure{code=<0x55920bc993b0>,arity=1,env=(int(1000))})
double(3.140000)
variant0(double(2.180000),string("a"))
variant0(
  object{class=string("Dune__exe__Example.U"),object-id=int(11)},
  object{class=string("Dune__exe__Example.D"),object-id=int(9)})
variant0(
  object{class=string("Dune__exe__Example.U"),object-id=int(11)},
  variant0(
    object{class=string("Dune__exe__Example.E"),object-id=int(10)},
    int(10)))
lazy(closure{code=<0x55920bc99420>,arity=1,env=()})
forward(int(3))
forcing(<0x1>)
closure{
  code=<0x55920bc99450>,
  arity=1,
  _=<0xcf9>,
  _=<0x55920bc99470>,
  _=<0x100000000000005>,
  env=()}
infix{
  offset=-24,
  whole=
    closure{
      code=<0x55920bc99450>,
      arity=1,
      _=<0xcf9>,
      _=<0x55920bc99470>,
      _=<0x100000000000005>,
      env=()}}
variant0(int(1),int(2),int(3))
double_array(1.,2.,3.)
variant0(int(1),variant0(int(2),variant0(int(3),int(0))))
cont(<0x55920d996b11>)
